src/controllers/menu.controller.ts

```diff
@@ -0,0 +1,82 @@
+import * as Koa from 'koa'
+import {getRepository, Repository} from 'typeorm'
+import Dish from '../models/menu.entity'
+import * as HttpStatus from 'http-status-codes'
+import {Like} from "typeorm";
+
+export default class MenuController {
+
+    static async getMenu(ctx: Koa.Context): Promise<void> {
+        const menuRepo: Repository<Dish> = getRepository(Dish)
+        const DishName = ctx.request.query.name
+        const CategoryName = ctx.request.query.category
+        let dishes: Dish[]
+        if (ctx.request.query.name) {
+            dishes = await menuRepo.find(
+                {
+                    name: Like(`%${(DishName as string).toUpperCase()}%`)
```

Comment:
As I understand you need `DishName as string` cast just because typescript thinks that you can have undefined here. This is because it doesn't understand that `if (ctx.request.query.name)` and `DishName` thing is the same data. What you need to do is either change `if` condition to use `DishName` or change here to use query parameter, but you shouldn't have casting like this, because it breaks the whole idea of using typescript for checking the correctness of your code.